tests/integration/components/sidenote-test.js

```diff
@@ -0,0 +1,161 @@
+        @offsetY={{this.note.offsetY}}
+        @data={{this.note.data}}
+        @onRegister={{this.onRegister}}
+        @onResize={{this.onResize}}
+        @onSelect={{this.onSelect}}
+        @onUnselect={{this.onUnselect}}
+      />
+    `);
+  });
+
+  test('it expose isActive with correct value', async function (assert) {
+    this.set('selectedSidenoteId', this.note.id);
+
+    await render(hbs`
+      <Sidenote
+        @id={{this.note.id}}
+        @onRegister={{this.onRegister}}
+        @onResize={{this.onResize}}
+        @onSelect={{this.onSelect}}
+        @onUnselect={{this.onUnselect}}
+        @selectedSidenoteId={{this.selectedSidenoteId}}
+        as |s|
+      >
+        {{#if s.isActive}}
+          <div data-active-sidenote>This is some dummy text.</div>
+        {{/if}}
+      </Sidenote>
+    `);
+
+    assert.dom('[data-active-sidenote]').exists('isActive is exposed as true');
+
+    this.set('selectedSidenoteId', -1);
+
+    assert
+      .dom('[data-active-sidenote]')
+      .doesNotExist('isActive is exposed as false');
+  });
+
+  test('it triggers onResize when element size change', async function (assert) {
+    this.setProperties({
+      showMore: false,
+      onResizeCalled: false,
+      onResize: () => {
+        this.set('onResizeCalled', true);
+      },
+    });
+
+    await render(hbs`
+      <Sidenote
+        @id={{this.note.id}}
+        @onRegister={{this.onRegister}}
+        @onResize={{this.onResize}}
+        @onSelect={{this.onSelect}}
+        @onUnselect={{this.onUnselect}}
+      >
+        {{#if this.showMore}}
+          <div data-show-more></div>
+        {{/if}}
+      </Sidenote>
+    `);
+
+    this.set('showMore', true);
+    await waitUntil(() => this.onResizeCalled);
+    assert.ok(this.onResizeCalled, 'onResize has been called');
+
+    assert.dom('[data-show-more]').exists('element is showed');
+
+    this.set('showMore', false);
+    await waitUntil(() => this.onResizeCalled);
+    assert.ok(this.onResizeCalled, 'onResize has been called');
+
+    assert.dom('[data-show-more]').doesNotExist('element is not showed');
+  });
+
+  test('it calls onSelect and onUnselect when using exposed actions', async function (assert) {
+    assert.expect(2);
+    const onSelectDone = assert.async();
+    const onUnselectDone = assert.async();
+
+    this.setProperties({
+      onSelect: () => {
+        assert.ok(true, 'onSelect is called');
+        onSelectDone();
+      },
+      onUnselect: () => {
+        assert.ok(true, 'onUnselect is called');
+        onUnselectDone();
+      },
+    });
+
+    await render(hbs`
+      <Sidenote
+        @id={{this.note.id}}
+        @onRegister={{this.onRegister}}
+        @onResize={{this.onResize}}
+        @onSelect={{this.onSelect}}
+        @onUnselect={{this.onUnselect}}
+        as |s|
+      >
+        <button
+          data-select-button
+          type="button"
+          {{on 'click' s.select}}
+        ></button>
```

Comment:
Even though it's a test please add labels to the buttons pray

Reply:
https://github.com/concordnow/ember-sidenotes/pull/1/commits/e33eeff28303467c451740e10d7d030ce0a4ab0f